cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/NativeDependencyResolverImpl.java

```diff
@@ -0,0 +1,312 @@
+import org.wso2.ballerinalang.compiler.util.CompilerContext;
+import org.wso2.ballerinalang.compiler.util.ProjectDirs;
+import org.wso2.ballerinalang.programfile.ProgramFileConstants;
+import org.wso2.ballerinalang.util.RepoUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+
+import static org.ballerinalang.tool.LauncherUtils.createLauncherException;
+import static org.wso2.ballerinalang.compiler.util.ProjectDirConstants.BALO_PLATFORM_LIB_DIR_NAME;
+import static org.wso2.ballerinalang.compiler.util.ProjectDirConstants.BLANG_COMPILED_JAR_EXT;
+import static org.wso2.ballerinalang.compiler.util.ProjectDirConstants.BLANG_PKG_DEFAULT_VERSION;
+import static org.wso2.ballerinalang.compiler.util.ProjectDirConstants.DIST_BIR_CACHE_DIR_NAME;
+
+/**
+ * Resolves jars and native libs for modules.
+ *
+ * @since 1.3.0
+ */
+public class NativeDependencyResolverImpl implements NativeDependencyResolver {
+    private static final CompilerContext.Key<NativeDependencyResolver> JAR_RESOLVER_KEY = new CompilerContext.Key<>();
+    private List<String> supportedPlatforms = Arrays.stream(ProgramFileConstants.SUPPORTED_PLATFORMS)
+            .collect(Collectors.toList());
+    private final BuildContext buildContext;
+    private final Path sourceRootPath;
+    private final String balHomePath;
+    private final Manifest manifest;
+    private boolean skipCopyLibsFromDist;
+
+    public static NativeDependencyResolver getInstance(BuildContext buildContext, boolean skipCopyLibsFromDist) {
+        CompilerContext context = buildContext.get(BuildContextField.COMPILER_CONTEXT);
+        NativeDependencyResolver nativeDependencyResolver = context.get(JAR_RESOLVER_KEY);
+        if (nativeDependencyResolver == null) {
+            nativeDependencyResolver = new NativeDependencyResolverImpl(buildContext, skipCopyLibsFromDist);
+        }
+        context.put(JAR_RESOLVER_KEY, nativeDependencyResolver);
+        return nativeDependencyResolver;
+    }
+
+    private NativeDependencyResolverImpl(BuildContext buildContext, boolean skipCopyLibsFromDist) {
+        CompilerContext context = buildContext.get(BuildContextField.COMPILER_CONTEXT);
+        this.buildContext = buildContext;
+        this.manifest = ManifestProcessor.getInstance(context).getManifest();
+        this.sourceRootPath = buildContext.get(BuildContextField.SOURCE_ROOT);
+        this.skipCopyLibsFromDist = skipCopyLibsFromDist;
+        this.balHomePath = buildContext.get(BuildContextField.HOME_REPO).toString();
+        supportedPlatforms.add("any");
```

Comment:
```suggestion
        supportedPlatforms.add(ProgramFileConstants.ANY_PLATFORM);
```
Can we use this content without adding a String here?